src/hrc721.spec.ts

```diff
@@ -35,158 +38,314 @@ describe('HRC721 Contract Interface', () => {
     expect(WALLET_PROVIDER_TEST_1).to.not.be.undefined
   })
 
-  describe.skip('balanceOf', () => {
+  describe('balanceOf', () => {
     it('should get the number of tokens in the specified account', async () => {
-      const balance = await contract.balanceOf(TEST_ADDRESS_1, TX_OPTIONS)
+      const mockedBalance = await FAKE_BALANCE_HRC721
+      const stub = sinon.stub(contract, 'call').withArgs('balanceOf', [TEST_ADDRESS_1], TX_OPTIONS)
+      stub.resolves().returns(FAKE_BALANCE_HRC721)
 
-      expect(balance).to.not.be.null
-      expect(balance).to.not.be.undefined
-      expect(balance).to.be.an.instanceof(BN)
-      expect(balance.gt(new BN(0))).to.be.true
+      const call = await contract.call<BN>('balanceOf', [TEST_ADDRESS_1], TX_OPTIONS)
+
+      expect(stub.calledOnce).to.be.true
+      expect(stub.callCount).to.be.equals(1)
+      expect(call).to.be.equals(mockedBalance)
     })
 
     it('should throw an error if address is not provided', async () => {
-      expect(contract.balanceOf('')).to.be.rejectedWith(Error)
+      const stub = sinon.stub(contract, 'call')
+      stub.withArgs('balanceOf', [], TX_OPTIONS).onFirstCall().rejects()
+
+      expect(contract.call<BN>('balanceOf', [], TX_OPTIONS)).to.be.rejectedWith(Error)
```

Comment:
same as above, here you are using `call` directly, but we should be testing the balanceOf method

Reply:
Got it!